### PR TITLE
convert inc updates to pytest

### PR DESCRIPTION
**Pull request by swadeley**
Convert long run incremental update checks to pytest.

These tests used two VMs, two RHEL repos, and a HC for the test and took between 942.31s (0:15:42) and 1729.99s (0:28:49).
I have simplified the set-up and now it takes  528.15s (0:08:48).

Thank you

### CODE REVIEW

**Comment by swadeley** [comment]
trigger: test-robottelo
pytest: tests/foreman/longrun/test_inc_updates.py

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/longrun/test_inc_updates.py
=========================== test session starts =============================================
platform linux -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /robot-test/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, cov-2.11.1, mock-3.5.1, ibutsu-1.14.1, reportportal-5.0.8, xdist-2.2.1
collected 2 items                                                                                                                                                                                 

tests/foreman/longrun/test_inc_updates.py ..                                                                                                                                                [100%]
======================== 2 passed, 8 warnings in 488.69s (0:08:08) ===========================

File: tests/foreman/longrun/test_inc_updates.py [hunk 1]
```diff
@@ -16,279 +16,240 @@
...
+        .search(
+            query={'search': f'name={ENVIRONMENT} and organization_id={module_manifest_org.id}'}
+        )[0]
+        .read()
+    )
+
+
+@pytest.fixture(scope='module')
+def dev_lce(module_manifest_org):
+    dev_lce = entities.LifecycleEnvironment(name='DEV', organization=module_manifest_org).create()
```
Comment (mshriver) [inline tests/foreman/longrun/test_inc_updates.py:54]: Consider just one-lining methods like these to directly return the instance that's created. Totally optional, just no use in creating a variable name that's only used once in the scope.

File: tests/foreman/longrun/test_inc_updates.py [hunk 2]
```diff
@@ -16,279 +16,240 @@
...
+def host(
+    module_manifest_org,
+    dev_lce,
+    qe_lce,
+    custom_repo,
+    module_ak,
+    module_cv,
+):
+    # Create client machine and register it to satellite with rhel_7_partial_ak
+    with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as vm:
```
Comment (mshriver) [inline tests/foreman/longrun/test_inc_updates.py:142]: Nice work including the VMBroker migration here!

File: tests/foreman/longrun/test_inc_updates.py [hunk 3]
```diff
@@ -16,279 +16,240 @@
...
+    module_cv = entities.ContentView(
+        organization=module_manifest_org, repository=[rhel7_sat6tools_repo.id, custom_repo.id]
+    ).create()
+    module_cv.publish()
+    module_cv = module_cv.read()
+    return module_cv
+
+
+@pytest.fixture(scope='module')
+def module_ak(module_manifest_org, module_cv, custom_repo, module_lce_library):
```
Comment (mshriver) [inline tests/foreman/longrun/test_inc_updates.py:104]: Great fixture structure, layering the dependent entities.

**Review by mshriver — APPROVED** [review]

File: tests/foreman/longrun/test_inc_updates.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py:54]: ACK

File: tests/foreman/longrun/test_inc_updates.py (hunk 2 above)
Comment (JacobCallahan) [reply tests/foreman/longrun/test_inc_updates.py:142]: He could also use one of the content host fixtures

File: tests/foreman/longrun/test_inc_updates.py [hunk 4]
```diff
@@ -16,279 +16,240 @@
...
-            }
-        )
-
-        # Re-read the content view to get the latest versions
-        self.rhel_6_partial_cv = self.rhel_6_partial_cv.read()
-        self.assertGreater(len(self.rhel_6_partial_cv.version), len(cv_versions))
+        # Add filter of type include but do not include anything.
+        # this will hide all RPMs from selected erratum before publishing.
+        entities.RPMContentViewFilter(
+            content_view=module_cv, inclusion=True, name="Include Nothing"
```
Comment (JacobCallahan) [inline tests/foreman/longrun/test_inc_updates.py:157]: Please switch these to single-quote strings.

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/longrun/test_inc_updates.py [hunk 5]
```diff
@@ -16,279 +16,240 @@
...
-        # Find the content host and ensure that tasks started by package
-        # installation has finished
-        host = entities.Host().search(query={'search': f'name={self.vm.hostname}'})
-        wait_for_tasks(
-            search_query='label = Actions::Katello::Host::UploadPackageProfile'
-            ' and resource_id = {}'
-            ' and started_at >= "{}"'.format(host[0].id, timestamp)
-        )
+from robottelo.constants.repos import FAKE_9_YUM_REPO
+from robottelo.hosts import ContentHost
```
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py:35]: its a pity, but my syntax checker cant cope with fixtures: E0602 undefined name 'ContentHost' [pyflakes]

File: tests/foreman/longrun/test_inc_updates.py [hunk 6]
```diff
@@ -16,279 +16,240 @@
 
 :Upstream: No
 """
-from datetime import date
-from datetime import datetime
-from datetime import timedelta
-
 import pytest
-from fauxfactory import gen_alpha
+from broker.broker import VMBroker
```
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py:20]: ImportError: cannot import name 'VMBroker' from 'broker'

Do I need to update?

File: tests/foreman/longrun/test_inc_updates.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py:35]: # Create client machine and register it to satellite with rhel_7_partial_ak
>       with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as vm:
E       NameError: name 'ContentHost' is not defined

tests/foreman/longrun/test_inc_updates.py:140: NameError

[the remark this answers is not in the document]
Comment (mshriver) [reply tests/foreman/longrun/test_inc_updates.py:35]: Correct on the second one @swadeley.

You don't need to import the existing fixture functions in order to use them. Pytest will resolve the fixture name, first from the local class or module, then from any fixtures that are defined by conftest. Our contest file imports all of our fixture modules from `pytest_fixtures` directory, so all you have to do is add the fixture function name to the test function arguments in order to use it.  No need to re-define it locally with the test.

File: tests/foreman/longrun/test_inc_updates.py (hunk 4 above)
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: Done

File: tests/foreman/longrun/test_inc_updates.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: Done

File: tests/foreman/longrun/test_inc_updates.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py:35]: @mshriver yes, I know that, but I was not using that fixture. So the feedback is misleading as it does not focus on why I should not use with VMBroker as a session manager and rather just use content host fixture.

File: tests/foreman/longrun/test_inc_updates.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply tests/foreman/longrun/test_inc_updates.py:20]: yep. you should be at broker 0.1.13

File: tests/foreman/longrun/test_inc_updates.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply tests/foreman/longrun/test_inc_updates.py:35]: using the fixture reduces code duplication as well as sets a common place where modification to the host properties can be changed.

[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py:35]: @JacobCallahan I removed `with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as vm:` and rewrote that section using  `def host(rhel7_contenthost, ` but its function scope and I need module scope.

[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: I have added: `+@pytest.fixture(scope="module")
+def rhel7_contenthost_module():` and will use that.

**Comment by swadeley** [comment]
robottelo]$ pytest tests/foreman/longrun/test_inc_updates.py
============================== test session starts ================================================
platform linux -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /robot-test/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, cov-2.11.1, mock-3.5.1, ibutsu-1.14.1, reportportal-5.0.8, xdist-2.2.1
collected 2 items                                                                                                                                                                                 

tests/foreman/longrun/test_inc_updates.py ..                                                                                                                                                [100%]


========================= 2 passed, 8 warnings in 502.46s (0:08:22) ==================================

File: tests/foreman/longrun/test_inc_updates.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: he would have to add one first

File: tests/foreman/longrun/test_inc_updates.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: broker]$ broker --version
Version: 0.1.10

(broker) [broker]$ git pull
Already up to date.
(broker) [broker]$ git status
On branch master

File: tests/foreman/longrun/test_inc_updates.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply tests/foreman/longrun/test_inc_updates.py]: There should be a module-scoped version as well. If not, feel free to add one.

File: tests/foreman/longrun/test_inc_updates.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply tests/foreman/longrun/test_inc_updates.py]: From robottelo's directory: pip install -U -r requirements.txt

**Review by JacobCallahan — APPROVED** [review]
ACK, nice work!!

**Comment by swadeley** [comment]
thank you

File: tests/foreman/longrun/test_inc_updates.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: No requirements file in broker, so its just `pip install -U .`, and I still have `v`

[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/longrun/test_inc_updates.py]: Fixed now `Successfully installed broker-0.1.13`
After I made my last PR for broker, I changed back to master, but then forgot to specify `upstream master` when doing a git pull /o\

**Pull request merged**